src/components/select/select.tsx

```diff
@@ -295,6 +295,10 @@ export class Select {
     }
 
     private handleMenuTriggerKeyPress(event: KeyboardEvent) {
+        if (this.disabled || this.readonly) {
+            return;
+        }
+
```

Comment:
I believe this is treating the side-effect of the problem, rather than _fixing_ it the right way.

I guess the issue is that the trigger element is not a native HTML element such as  a`<button />` for instance, which could be used to perform an action (click, key-press, tap). Right now, it's a `div`. Therefore, it cannot get `disabled` for real. Currently it only has `aria-disabled` which doesn't really do much of anything.

If it was a `<button />` element instead, and would get the attribute of `disabled`, then we wouldn't need to add this special `if()` condition. The browser would rather take care of the correct disabled behavior for us.

<img width="549" alt="image" src="https://user-images.githubusercontent.com/35954987/202698344-a9c45f04-6a36-4249-acb6-eeeb70871456.png">

## considerations
1. Adding this will introduce some visual bugs that we inherit from the user agent stylesheets that must be fixed separately. Perhaps only below is enough 👇 but we may need even more fixes. So visually inspect it so that it looks exactly as before. <img width="285" alt="image" src="https://user-images.githubusercontent.com/35954987/202699080-99b0a66b-b355-4a71-93f3-35015d35af06.png">
2. The[ MDC docs](https://m2.material.io/develop/web/components/input-controls/select-menus) tells us we should use a div for the trigger `<div class="mdc-select__anchor" aria-disabled="true">`. 
So it doesn't seem to be us, doing it the wrong way initially. 🤔 They also mention:
    > Add the mdc-select--disabled class to the mdc-select element, set the aria-disabled attribute on the mdc-select__selected-text element to be "true", and set the disabled attribute any hidden input element.

    and  also say:
    > NOTE: To programmatically set a select as disabled, use the disabled property in the MDCSelect API.

Therefore I am not really sure if using `<button` can create accessibility issues, or other problems for us? 
- Why do they say use `aria-disabled="true"`? 
- Why do their default example has `role="button"` on the anchor element,
   <img width="571" alt="image" src="https://user-images.githubusercontent.com/35954987/202702346-e9891f95-23df-45a0-9917-bc5a7aa52ae8.png">
    But their `disabled` example doesn't have it. 
    <img width="620" alt="image" src="https://user-images.githubusercontent.com/35954987/202702666-3c6e4191-2eb6-46d0-b3eb-0f2b7208fa89.png">
    Is that what we should remove instead of the above fix, when the component is disabled?
    
@adrianschmidt what do you think? 

*****
Additionally the native select element should get this:
<img width="545" alt="image" src="https://user-images.githubusercontent.com/35954987/202698548-20e37add-e18c-4b3b-8f4b-df58bdc41685.png">

Reply:
in other words, the fix should be this: 
> Don't allow the trigger be focus-able using tabbing, when `readonly` or `disabled`.

Not this:
> Disable `handleMenuTriggerKeyPress`, when `readonly` or `disabled`. (this fix)


We could perhaps simply put `tabindex="-1"` on the trigger element. 🤔
This way, we don't need to that other commit for the `helperText`.

Reply:
Yes, it seems like a better solution. I have pushed these changes, and if Adrian doesn't object to them, I guess, we can proceed. 
Also, I couldn't notice any weird behavior or different styling after testing it.

Reply:
Cool. regarding accessibility, I think this change should be completely fine. 
I was reading on `aria-expanded` that is on the trigger element, which for us is always on `false`. This is another accessibility bug that we should fix: https://github.com/Lundalogik/lime-elements/issues/1966. However, reading on that, I saw some examples: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-expanded and it seems this change is better than what MDC suggests. Super strange to me.

Reply:
I think you're right about using button (at least if using `role="button"` was right to begin with). From what I understand, from accessibility point of view, `<div role="button">` is just the same thing as `<button>`

Disabling focus on a div seems to me like unneeded work to reinvent the button behaviour.